File: pdu/python/geometry_msgs/pdu_cdr_conv_Pose2D.py
from .pdu_pytype_Pose2D import Pose2D
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_Pose2D(writer: CdrWriter, src: Pose2D):
    writer.write_float64(src.x)
    writer.write_float64(src.y)
    writer.write_float64(src.theta)


def cdr_body_to_py_Pose2D(reader: CdrReader, dst: Pose2D):
    dst.x = reader.read_float64()
    dst.y = reader.read_float64()
    dst.theta = reader.read_float64()
    return dst


def py_to_cdr_Pose2D(src: Pose2D) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_Pose2D(writer, src)
    return writer.bytes()


def cdr_to_py_Pose2D(cdr_payload) -> Pose2D:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = Pose2D()
    return cdr_body_to_py_Pose2D(reader, dst)

File: pdu/javascript/geometry_msgs/pdu_cdr_conv_Pose2D.js
import { Pose2D } from './pdu_jstype_Pose2D.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduPose2DConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { Pose2D } src
     */
    static to_cdr_body(writer, src) {
        writer.write_float64(src.x);
        writer.write_float64(src.y);
        writer.write_float64(src.theta);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { Pose2D } dst
     * @returns { Pose2D }
     */
    static cdr_body_to_js(reader, dst) {
        dst.x = reader.read_float64();
        dst.y = reader.read_float64();
        dst.theta = reader.read_float64();
        return dst;
    }

    /**
     * @param { Pose2D } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { Pose2D }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new Pose2D());
    }
}
